src/main/java/com/example/java/maven/game2048/Game.java

```diff
@@ -7,9 +7,74 @@
     public Game(int winningNumber, Board board) {
         this.winningNumber = winningNumber;
         this.board = board;
+        board.placeTwoInitialElementsOnBoard();
+        MessagePrinter.printInstructionsBeforeGame();
     }
 
+
     public void start() {
+        while (!board.checkIfGameOver()) {
+            System.out.println(board);
+            makeValidMove();
+            if (board.checkIfWon(winningNumber)) {
+                break;
+            }
+            board.placeNextElementOnBoard();
+        }
+    }
+
+    private String getPlayerMove() {
+        String userInput = UserInputProvider.getMoveFromPlayer();
+        if (playerMoveIsValid(userInput)) {
+            return userInput;
+        }
+        return getPlayerMove();
+    }
+
+    private boolean playerMoveIsValid(String userInput) {
+        if (userInput.equalsIgnoreCase("a")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("d")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("w")) {
+            return true;
+        }
+        return userInput.equalsIgnoreCase("s");
+    }
+
+    private void makeValidMove() {
```

Comment:
Ok. Correct me if I am wrong. Current logic of move looks like (example for `moveLeft`):
```
//given
row = [0,0,2,2,4,8,0,2,4,8];

//when
moveLeft()

//then
row = [2,2,4,8,2,4,8,0,0,0];

//when
sumLeft()

//then
row = [4,0,4,8,2,4,8,0,0,0];

//when
moveLeft()

//then
row = [4,4,8,2,4,8,0,0,0,0];
```
Am I right? :D

Reply:
Yes